src/components/custom/filter/filter/filter.tsx

```diff
@@ -7,13 +8,16 @@ export const Filter = (props: IFilterProps) => {
 	// =============================================================================
 	// CONST, STATE, REF
 	// =============================================================================
+
+	const { reset } = useFormContext();
+
 	const {
 		id,
-		schema: { children, label, toggleFilterButtonLabel, onClear, clearButtonDisabled },
+		schema: { children, label, toggleFilterButtonLabel, clearButtonDisabled },
 	} = props;
 
 	const clearData = () => {
-		onClear && onClear();
+		reset();
```

Comment:
this will reset all the fields in the form, even those outside the filter component. there is a need to look through the children and reset only those fields within it.